server/sockets/cohort.js

```diff
@@ -0,0 +1,43 @@
+const { CohortActionTypes } = require('../common/variables');
+const Cohort = require('../models/cohort.model');
+const { responseWithCohort } = require('../common/utils');
+
+const addCohortMemberWS = (socket, cohortClients) => {
+  socket.on(CohortActionTypes.ADD_MEMBER_SUCCESS, data => {
+    const { cohortId } = data;
+
+    // sends new cohort member to all user on this cohort view
+    socket.broadcast
+      .to(`cohort-${data.cohortId}`)
+      .emit(CohortActionTypes.ADD_MEMBER_SUCCESS, data);
+
+    // sends updated cohort metadata to users (which are members
+    // of this cohort including the new one) on cohorts view
+    if (cohortClients.size > 0) {
+      Cohort.findById(cohortId)
+        .select('_id isArchived createdAt name description memberIds')
+        .lean()
+        .exec()
+        .then(doc => {
+          if (doc) {
+            const { memberIds } = doc;
+            const cohort = responseWithCohort(doc);
+
+            memberIds.forEach(id => {
+              const memberId = id.toString();
+
+              if (cohortClients.has(memberId)) {
+                socket.broadcast
```

Comment:
please check if needed

Reply:
Theoretically, it is not needed. All work fine without it. There is a small chance that if somebody adds a new user and will go quickly to the cohorts view or cohort's shared list's view and the backend will perform its duty slowly, they probably receive data via socket also. But even if it happens, it won't cause an error because the received data will overwrite the existing one. So I decided to remove it.